### PR TITLE
Add verify

### PR DESCRIPTION
**Pull request by dianabarsan**
Add verify command which: 
- does sync_shards for every DB
- does an _all_docs request (with limit 0
- checks view indexes (returning results on a stale request is 👍🏻 ) 

Caveat for views: if the view has no docs to index (none of the docs emit in the view), we will get false failures, so on no doc views we just display a warning. Wording of the warning can definitely be improved.

### CODE REVIEW

File: src/verify.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    const views = Object.keys(ddoc.views);
+    for (const view of views) {
+      viewsChecked = true;
+      const url = utils.getUrl(`${dbName}/${ddoc._id}/_view/${view}`, false, 'stale=ok&limit=0');
+      const viewResponse = await utils.request({ url });
+
+      if (viewResponse.total_rows > 0) {
+        // at least one view being indexed means that the .shard folders for the database were moved successfully
+        // some views might not index any documents
+        viewsIndexed = true;
```
Comment (garethbowen) [inline src/verify.js]: Why not short circuit here? ie...

```suggestion
        return true;
```

File: src/verify.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+const verifyViews = async (dbName, numDocs) => {
+  if (!numDocs) {
+    return true;
+  }
+
+  const url = utils.getUrl(`${dbName}/_design_docs`, false, 'include_docs=true');
+  const response = await utils.request({ url });
+
+  let viewsIndexed = false;
+  let viewsChecked = false;
```
Comment (garethbowen) [inline src/verify.js]: If I'm reading this correctly, the only time this will stay false is if, there are no ddocs or no views, right? It would be more readable to short circuit this special case, eg:

```suggestion
  const ddocsWithViews = response.rows.filter(row => row.ddoc && row.ddoc.views && row.ddoc.views.length !== 0);
  if (!ddocsWithViews.length) {
    return true; // nothing to verify
  }
  for (const { doc: ddoc } of ddocsWithViews) {
```

Obviously I've just hacked this together - ignore typos and obvious bugs!

If you do this and the other short circuit mentioned below you can drop both variables and just `return false` at the end, which makes the code easier to follow because you don't need to track all the ways in which the variable can change.

**Review by garethbowen — CHANGES_REQUESTED** [review]
Awesome! I found the `verify` function hard to follow but I've put some suggestions that I think make it easier.

File: src/verify.js (hunk 1 above)
Comment (dianabarsan) [reply src/verify.js]: I did short circuit initially. Then I wanted to test all views (I did some manual custom tests, for good measure) and forgot to revert the code back.

Comment (dianabarsan) [reply src/verify.js]: Ah, I can see that I later added unit tests to check I go over every view too!

File: package.json [hunk 3]
```diff
@@ -17,9 +17,10 @@
     "move-node": "bin/move-node.js",
     "move-shard": "bin/move-shard.js",
     "move-shards": "bin/move-shards.js",
-    "remove-node": "bin/remove-node.js",
+    "remove-node": "bin/remove-node.js",
```
Comment (garethbowen) [inline package.json]:
```suggestion
    "remove-node": "bin/remove-node.js",
```

File: test/unit/verify.spec.js [hunk 4]
```diff
@@ -0,0 +1,196 @@
+const utils = require('../../src/utils');
+const verify = require('../../src/verify');
+
+const couchUrl = utils.couchUrl.toString();
+
+describe('verify', () => {
+  describe('verity', () => {
```
Comment (garethbowen) [inline test/unit/verify.spec.js]: Unnecessary double describe!

**Review by garethbowen — APPROVED** [review]
Awesome. Two optional nits inline.

File: package.json (hunk 3 above)
Comment (dianabarsan) [reply package.json]: UGH I pay this price when  I resolve conflicts in GitHub UI...
